WordPress/gutenberg
# Docs: Consolidate React Native documentation

## Description

Update React Native documentation to consolidate into one section under the Contributors Guide. This PR also gathers up the documentation under a new `react-native` folder within the code section of the contributor's guide.

This organizes the docs together into one section, organizing the side bar better. Plus moves the setup for OSX into the contributor's section since contributors are the most likely developers this avoid some confusion that arose from a developer looking to extend WordPress / Gutenberg.


## How has this been tested?

- Confirm new links are correct
- Search code base for previous links are updated


## Types of changes

- Adds new react-native folder and moves docs to it
- Updates table of contents
- Update links to new react-native locations
- Minor rewording for consistency
